src/skin.cpp

```diff
@@ -111,9 +112,32 @@ void generateSkinAreas(int layer_nr, SliceMeshStorage& mesh, const int innermost
                 upskin = upskin.difference(not_air); // skin overlaps with the walls
             }
         }
+
+        int expand_skins_expand_distance = mesh.getSettingInMicrons("expand_skins_expand_distance");
         
+        if (expand_skins_expand_distance > 0)
+        {
+            int expand_skins_shrink_distance = mesh.getSettingInMicrons("expand_skins_shrink_distance");
+
+            // skin areas are to be enlarged by expand_distance but before they are expanded
+            // the skin areas are shrunk by shrink_distance so that very narrow regions of skin
+            // (often caused by the model's surface having a steep incline) are removed first
+
+            expand_skins_expand_distance += expand_skins_shrink_distance; // increase the expansion distance to compensate for the shrinkage
+
+            if (mesh.getSettingBoolean("expand_upper_skins"))
+            {
+                upskin = upskin.offset(-expand_skins_shrink_distance).offset(expand_skins_expand_distance).intersection(virgin_upskin);
```

Comment:
Those people only want to have small skin removed when it's next to infill. When the skin is enclosed directly by the walls, you would be removing the actual skin, which is a bad idea. You would then be able to see the infill when looking at your print.

I think the correct place to fix the issue you refer to is when the skins are generated in the first place. I'm not 100% sure it's possible there, but If it's possible we should do it there, because that would be less of an impact on performance.